### PR TITLE
Fix search/downloads

### PR DESCRIPTION
**Pull request by gstcyr**
Still see the odd weird behavior but seems to work well for the most part.

Add more command line tools for testing

### CODE REVIEW

File: libgen_client.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/env python3
+import getopt
```
Comment (fallaciousreasoning) [inline libgen_client.py]: Could you move this import into `main` so the plugin still works if `getopt` isn't installed?

File: libgen_client.py [hunk 2]
```diff
@@ -159,11 +161,43 @@ def get_download_url(self, md5):
...
+            language = arg
+        elif opt in ("-f", "--format"):
+            file_format = arg
+
+    search_results = client.search(query, criteria, language, file_format)
+
 
     for result in search_results.results[:5]:
-        print(result.title)
+        print(result.title+" by "+result.authors)
```
Comment (fallaciousreasoning) [inline libgen_client.py]: Nit: spacing
```suggestion
        print(result.title + " by " + result.authors)
```

File: libgen_client.py [hunk 3]
```diff
@@ -159,11 +161,43 @@ def get_download_url(self, md5):
             except:
                 continue
 
-if __name__ == "__main__":
+def main(argv):
     client = LibgenFictionClient()
-    search_results = client.search("the count of monte cristo")
+
+    opts, args = getopt.getopt(argv, "hq:t:a:s:l:f:", ["query=", "title=", "author=", "series=", "language=", "format="])
```
Comment (fallaciousreasoning) [inline libgen_client.py]: Personally, I prefer `argparse` for parsing arguments, but this is fine :smile:

File: libgen_client.py [hunk 4]
```diff
@@ -38,17 +40,17 @@ def __init__(self, title, authors, series, md5, mirrors, language,
...
         SERIES_XPATH = '/td[2]'
-        TITLE_XPATH = '/td[3]//a'
+        TITLE_XPATH = '/td[3]/p/a'
         LANGUAGE_XPATH = '/td[4]'
         FILE_XPATH = '/td[5]'
         MIRRORS_XPATH = '/td[6]//a'
 
         # Parse the Author(s) column into `authors`
-        authors = ' & '.join([
+        authors = ' & '.join(filter(None, [
```
Comment (fallaciousreasoning) [inline libgen_client.py:47]: Can this `filter` be removed?

**Review by fallaciousreasoning — CHANGES_REQUESTED** [review]
Mostly looks pretty good to me, thanks!

File: libgen_client.py (hunk 4 above)
Comment (gstcyr) [reply libgen_client.py:47]: I was getting random errors without it

**Review by fallaciousreasoning — APPROVED** [review]
LGTM

**Pull request merged**